src/schema/v2/artwork/index.ts

```diff
@@ -370,6 +377,53 @@ export const ArtworkType = new GraphQLObjectType<any, ResolverContext>({
         description:
           'Represents the "**medium type**", such as _Painting_. (This field is also commonly referred to as just "medium", but should not be confused with the artwork attribute called `medium`.)',
       },
+      collectionsConnection: {
+        type: CollectionsConnectionType,
+        args: pageable({
+          page: { type: GraphQLInt },
+          size: { type: GraphQLInt },
+          default: { type: GraphQLBoolean },
+          saves: { type: GraphQLBoolean },
+          sort: { type: CollectionSorts },
+        }),
+        resolve: async (parent, args, context, _info) => {
+          const { id: artwork_id } = parent
+          const { collectionsLoader, userID: meID } = context
+          if (!collectionsLoader) return null
+
+          const { page, size, offset } = convertConnectionArgsToGravityArgs(
+            args
+          )
+
+          const { body, headers } = await collectionsLoader({
+            artwork_id,
+            user_id: meID,
+            private: true,
+            default: args.default,
+            saves: args.saves,
+            sort: args.sort,
+            size,
+            offset,
+            total_count: true,
+          })
+          const bodyWithMe = (body ?? []).map((obj) => {
+            return { ...obj, userID: meID }
+          })
+          const totalCount = parseInt(
+            (headers ?? {})["x-total-count"] || "0",
+            10
+          )
```

Comment:
**question**: I think `headers` will never be `null` or `undefined`. Or am I wrong?

Reply:
**suggestion**: small improvements

```javascript
const totalCount = parseInt(headers?.["x-total-count"] || "0", 10)
```